pkg/volume/hugepages/hugepages.go

```diff
@@ -0,0 +1,304 @@
+package hugepages
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/util/mount"
+	strutils "k8s.io/kubernetes/pkg/util/strings"
+	"k8s.io/kubernetes/pkg/volume"
+	volumeutil "k8s.io/kubernetes/pkg/volume/util"
+)
+
+// http://issue.k8s.io/2630
+const perm os.FileMode = 0777
+
+// ProbeVolumePlugins is the primary entrypoint for volume plugins.
+func ProbeVolumePlugins() []volume.VolumePlugin {
+	return []volume.VolumePlugin{
+		&hugePagesPlugin{nil},
+	}
+}
+
+type hugePagesPlugin struct {
+	host volume.VolumeHost
+}
+
+var _ volume.VolumePlugin = &hugePagesPlugin{}
+
+const (
+	hugePagesPluginName  = "kubernetes.io/hugepages"
+	hugePagesTotalString = "HugePages_Total"
+)
+
+func getPath(uid types.UID, volName string, host volume.VolumeHost) string {
+	return host.GetPodVolumeDir(uid, strutils.EscapeQualifiedNameForDisk(hugePagesPluginName), volName)
+}
+
+func (plugin *hugePagesPlugin) Init(host volume.VolumeHost) error {
+	plugin.host = host
+	return nil
+}
+
+func (plugin *hugePagesPlugin) GetPluginName() string {
+	return hugePagesPluginName
+}
+
+func (plugin *hugePagesPlugin) GetVolumeName(spec *volume.Spec) (string, error) {
+	if volumeSource, _ := getVolumeSource(spec); volumeSource == nil {
+		return "", fmt.Errorf("Spec does not reference a HugePages volume type")
+	}
+	// Return user defined volume name
+	return spec.Name(), nil
+}
+
+var readFile = ioutil.ReadFile
+
+func detectHugepages() error {
```

Comment:
return the amount of huge pages too and validate max is not more than huagepages_free or hugepages_total.